tooling/bin/assess-minibroker-charts.rb

```diff
@@ -0,0 +1,622 @@
+#!/usr/bin/env ruby
+##
+# The purpose of this script is to assess which of the publicly
+# available charts for the databases supported by minibroker will work
+# with SCF.
+
+# Configuration
+# - Fixed location: `stable` helm repository
+# - Configurable:   work directory for state
+# - Configurable:   SCF namespace
+# - Configurable:   Cluster Admin Password
+# - Configurable:   Operation mode (full, incremental)
+
+require 'fileutils'
+require 'net/http'
+require 'open3'
+require 'optparse'
+require 'uri'
+require 'yaml'
+
+# Global option for error handling in run, capture
+$opts = { errexit: true }
+
+@top = File.dirname(File.dirname(File.dirname(File.absolute_path(__FILE__))))
+
+def main
+  config
+  base_statistics
+
+  @assessed = 0
+  @skipped = 0
+  state
+
+  engines.each do |engine|
+    master_index[engine].each do |chart|
+      enginev       = chart['appVersion']
+      chartv        = chart['version']
+      chartlocation = chart['urls'].first
+
+      # We are ignoring all the entries for which we do not have the
+      # engine version. Because that is the plan id later, therefore
+      # required.
+      next unless enginev
+
+      next if skip_chart(engine, enginev, chartv)
+      assess_chart(chart, engine, enginev, chartv, chartlocation)
+      @assessed += 1
+    end
+  end
+
+  rewind
+  if @skipped
+    puts "#{"Skipped".cyan}:  #{@skipped}"
+  end
+  if @assessed
+    puts "#{"Assessed".cyan}: #{@assessed}"
+  end
+end
+
+def config
+  @workdir     = File.join(@top, '_work/mb-chart-assessment')
+  @namespace   = "cf"
+  @auth        = "changeme"
+  @incremental = false
+
+  OptionParser.new do |opts|
+    opts.banner = "Usage: assess-minibroker-charts [options]"
+    opts.on("-wPATH", "--work-dir=PATH", "Set work directory for state and transients") do |v|
+      @workdir = v.to_s
+    end
+    opts.on("-nNAME", "--namespace=NAME", "Set SCF namespace") do |v|
+      @namespace = v.to_s
+    end
+    opts.on("-pPASS", "--password=PASS", "Set cluster admin password") do |v|
+      @auth = v.to_s
+    end
+    opts.on("-i", "--incremental", "Activate incremental mode") do |v|
+      @incremental = true
+    end
+  end.parse!
+
+  puts "Configuration".cyan
+  puts "  - Namespace: #{@namespace.blue}"
+  puts "  - Password:  #{@auth.blue}"
+  puts "  - Mode:      #{mode.blue}"
+  puts "  - Top:       #{@top.blue}"
+  puts "  - Work dir:  #{@workdir.blue}"
```

Comment:
Do we need to ensure that `@workdir` exists?

Reply:
Better yes. My testing was always with it existing, so not sure without. done, local.